cmd/provisioner-localpv/app/helper.go

```diff
@@ -89,20 +91,18 @@ func (p *Provisioner) createCleanupPod(cmdsForPath []string, name, path, node st
 		return fmt.Errorf("invalid empty name or path or node")
 	}
 
-	//Validate that non-root directories are not passed for delete
-	// and also perform white/black list validations.
-	config := &VolumeConfig{}
-	path, err = config.validatePath(path)
+	// Initialize HostPath builder and validate that
+	// non-root directories are not passed for delete
+	hostPathBuilder := mHostPath.NewBuilder().WithPath(path).
+		WithCheckf(mHostPath.IsNonRoot(), "root directories are not allowed")
+
+	err = hostPathBuilder.Validate()
 	if err != nil {
 		return err
 	}
 
 	// Extract the base path and the volume unique path.
-	path = strings.TrimSuffix(path, "/")
-	parentDir, volumeDir := config.extractSubPath(path)
-	//parentDir, volumeDir := filepath.Split(path)
-	//parentDir = strings.TrimSuffix(parentDir, "/")
-	//volumeDir = strings.TrimSuffix(volumeDir, "/")
+	parentDir, volumeDir := hostPathBuilder.ExtractSubPath()
```

Comment:
Can `ExtractSubPath()` invoke `Validate()` internally?
If yes, then caller can simply invoke `ExtractSubPath()` that should return `string, string, error`

Reply:
Like I mentioned earlier, builder cannot be set with limiting rules. All are correct. 
What we can look instead is from the caller's side. Is caller logic more readable, concise yet achieves its purpose, & so on.